File: misc/encrypt_test.pl
use 5.018;

use lib qw(extlib/lib/perl5);
use lib qw(/home/void/hs/misc/extlib/lib/perl5);
use lib qw(/home/void/stock/extlib/lib/perl5);

use Try::Tiny;

use Crypt::CBC;
use Crypt::Cipher::AES;
use MIME::Base64;
use Digest::MD5 qw(md5);

use IO::Compress::Gzip qw(gzip $GzipError);
use IO::Uncompress::Gunzip qw(gunzip $GunzipError);

#my $status = gzip $input => $output or die "gzip failed: $GzipError\n";
#my $status = gunzip $input => $output or die "gunzip failed: $GunzipError\n";

my $data = "abcdde";
my $gziped;
gzip \$data => \$gziped or die "gzip failed: $GzipError\n";
my $password = 'random key';
my $key = md5($password);

my $cbc = Crypt::CBC->new(-cipher => 'Cipher::AES',
	-key => $key,
	-iv => pack('C*', map {0x00} 1..16),
	-header => 'none',
	-padding => 'standard',
	-keysize => 16,
	-literal_key => 1);
say encode_base64($cbc->encrypt($gziped));

my $data2 = '1Golj+yoVySpQSHXjBw7NjdBfw5u6pJAP8PYp0YcJjo=';
my $gunziped;
my $data3 = $cbc->decrypt(decode_base64($data2));
gunzip \$data3 => \$gunziped or die "gunzip failed: $GunzipError\n";
say $gunziped;

File: misc/encrypt_test.cpp
// g++ -std=c++11 -Wall encrypt_test.cpp hs_crypt.o tiny-AES128-C/md5.o tiny-AES128-C/aes.o hs_base64.o base64.o hs_compress.o -Luzlib/lib -ltinf

#include <iostream>
#include <string>

#include "hs_crypt.h"
#include "hs_compress.h"
#include "hs_base64.h"
#include "hs_exception.h"

int main()
{
	auto deflated = hs_gunzip(hs_decrypt(
				hs_base64_decode((const unsigned char *)"1Golj+yoVySpQSHXjBw7NjdBfw5u6pJAP8PYp0YcJjo=", 44),
				"random key"));
	std::cout << std::string(deflated.begin(), deflated.end()) << std::endl;

	const char *s = "abcdde";
	auto post_data = hs_base64_encode(
			hs_encrypt(
				hs_gzip((const unsigned char *)s, 6),
				"random key"));
	std::cout << std::string(post_data.begin(), post_data.end()) << std::endl;
}
